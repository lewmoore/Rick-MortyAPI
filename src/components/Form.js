import React from 'react';

class Form extends React.Component{
  render() {
    return (
      <form>
        <label>Character
        <input type='text' name='character' />
        <br/>
        </label>
        <input type='submit' value='search' />
      </form>
    )
  }
};

export default Form;
