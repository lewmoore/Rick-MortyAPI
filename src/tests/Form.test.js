import React from 'react';
import ReactDOM from 'react-dom';
import Form from '../components/Form';
import Enzyme from 'enzyme'
import { mount, shallow } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16'


Enzyme.configure({adapter: new EnzymeAdapter() });

describe('Form', () => {

  const form = shallow(<Form />);

  it("renders correctly", () => {
    expect(form).toMatchSnapshot();
  });
})
