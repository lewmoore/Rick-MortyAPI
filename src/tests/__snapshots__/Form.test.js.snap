// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
          Character
          <input
            name="character"
            type="text"
          />
          <br />
        </label>,
        <input
          type="submit"
          value="search"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Character",
            <input
              name="character"
              type="text"
            />,
            <br />,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Character",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "character",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "type": "submit",
          "value": "search",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            Character
            <input
              name="character"
              type="text"
            />
            <br />
          </label>,
          <input
            type="submit"
            value="search"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Character",
              <input
                name="character"
                type="text"
              />,
              <br />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Character",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "character",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "type": "submit",
            "value": "search",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
